src/common/Defs.h

```diff
@@ -61,9 +67,38 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define SERVERCONFIG_NAME   "autogen_server.cfg"
 
 #define UNNAMED_PLAYER      "UnnamedPlayer"
-#define UNNAMED_SERVER      PRODUCT_NAME " " PRODUCT_VERSION " Server"
 
 /** file containing our RSA public and private keys */
 #define RSAKEY_FILE        "pubkey"
 
+
+typedef struct {
+	const char name[25];
+	const char name_lower[25];
+	const char version[10];
+	const char default_base_pak[25];
+	const char* gamename_string;
+	const char* gamename_for_master;
+	const char* master_server;
+	const char* irc_server;
+	const char* irc_channel;
+	const char* www_baseurl;
+	const char* unnamed_server;
+} gameinfo_t;
+
+
+#define GAMES\
+	game(UNV)
+
+#define game(x) GAME_##x
+
+typedef enum { GAMES } Game_e;
+
+#undef game
+#define game(x) { PRODUCT_NAME_##x, PRODUCT_NAME_LOWER_##x, PRODUCT_VERSION_##x, DEFAULT_BASE_PAK_##x, GAMENAME_STRING_##x, GAMENAME_FOR_MASTER_##x, MASTER_SERVER_NAME_##x, IRC_SERVER_##x, IRC_CHANNEL_##x, WWW_BASEURL_##x, UNNAMED_SERVER_##x }
+
+const gameinfo_t gamesinfo [ ] = { GAMES };
```

Comment:
I really don't like the macro-based approach and the overall C-ness of gameinfo_t, I think it could lead to nasty issues and make it harder to extend in the future